XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Simulation/Hands/SimulatedHandController.cs

```diff
@@ -0,0 +1,163 @@
+                {
+                    return simulationHandControllerDataProvider.Profile;
+                }
+            }
+
+            return null;
+        }
+
+        /// <inheritdoc />
+        public override void UpdateController()
+        {
+            base.UpdateController();
+
+            // Read keyboard / mouse input to determine the root pose delta since last frame.
+            MixedRealityPose rootPoseDelta = new MixedRealityPose(
+                GetHandPositionDelta(), Quaternion.Euler(GetHandRotationDelta()));
+
+            // Calculate pose changes and compute timestamp for hand tracking update.
+            float poseAnimationDelta = profile.HandPoseAnimationSpeed * Time.deltaTime;
+            long timeStamp = handUpdateStopWatch.TimeStamp;
+
+            // Update simualted hand states using collected data.
+            HandState.Update(timeStamp, rootPoseDelta, GetTargetHandPose(), poseAnimationDelta);
+
+            lastMousePosition = Input.mousePosition;
+
+            UpdateBase(HandState.HandData);
+        }
+
+        /// <summary>
+        /// Gets a simulated Yaw, Pitch and Roll delta for the current frame.
+        /// </summary>
+        /// <returns>Updated hand rotation angles.</returns>
+        private Vector3 GetHandRotationDelta()
+        {
+            float rotationDelta = profile.RotationSpeed * Time.deltaTime;
+            Vector3 rotationDeltaEulerAngles = Vector3.zero;
+
+            if (Input.GetKey(profile.YawCCWKey))
+            {
+                rotationDeltaEulerAngles.y = -rotationDelta;
+            }
+            if (Input.GetKey(profile.YawCWKey))
+            {
+                rotationDeltaEulerAngles.y = rotationDelta;
+            }
+            if (Input.GetKey(profile.PitchCCWKey))
+            {
+                rotationDeltaEulerAngles.x = rotationDelta;
+            }
+            if (Input.GetKey(profile.PitchCWKey))
+            {
+                rotationDeltaEulerAngles.x = -rotationDelta;
+            }
+            if (Input.GetKey(profile.RollCCWKey))
+            {
+                rotationDeltaEulerAngles.z = rotationDelta;
+            }
+            if (Input.GetKey(profile.RollCWKey))
+            {
+                rotationDeltaEulerAngles.z = -rotationDelta;
+            }
+
+            return rotationDeltaEulerAngles;
+        }
+
+        /// <summary>
+        /// Gets a simulated depth tracking (hands closer / further from tracking device) update, as well
+        /// as the hands simulated (x,y) position.
+        /// </summary>
+        /// <returns>Hand movement delta.</returns>
+        private Vector3 GetHandPositionDelta()
+        {
+            Vector3 mouseDelta = (lastMousePosition.HasValue ? Input.mousePosition - lastMousePosition.Value : Vector3.zero);
+            mouseDelta.z += Input.GetAxis("Mouse ScrollWheel") * profile.HandDepthMultiplier;
```

Comment:
Is the hand position not customizable with other inputs? Only `ScrollWheel`?

I'd like to see all of these be turned into input actions if possible.

Reply:
So far only scroll wheel. Adding this to the Input Actions refactor PR.